File: chapter_1/str_num.py
def str_num(str):
    num = 0
    for c in str: 
        t = ord(c) - ord('0')
        num = 10*num+t
    print(f'beam {int(str,10)}')
    return num

if __name__ == '__main__':
    print(str_num("344"))

File: chapter_1/str_num.rs
fn str_to_num(s: &str) -> i16 {
    let num = s.chars().map(|c| {
        (c as u8) - ('0' as u8)
    }).inspect(|x|println!("beam {}",x))
    .fold(0,|mut acc, c| {
        if acc < i16::MAX/10 {
            acc = acc*10 + c as i16;
        } else {
            acc = i16::MAX;
        }
        acc
    });
    num
}

fn main() {
    let s = "334";
    println!("{}", str_to_num(s));
}